### ISSUE TITLE
bCNC on RPIzW

### ISSUE THREAD

**Issue by JLNapo** [vlachoudis/bCNC#1007]
Hello,
I have disconnections with the GRBL v1.1g card.
The RPI recreated a Serial / USB port (USB1 instead of USB0) because USB0 is no longer accessible.
I first thought that the disconnection was RPI too small for the task, but on my PC Windows 10 64bits it's the same, there are disconnections.
The problem is that I already broke several strawberries and I lost several circuits, so if you can look into the subject, it will arrange me well.
Ha, I forgot, I use bCNC v0.9.14 and I also often have errors #33.
The same GCode does not generate any error with Cambam and the GRBLMachine Plugin.
Thank you again for all your efforts.
Best regards

**Comment by JLNapo** [comment]
Hello,
Thank you for your answer.
Yes, it was a problem, but I was using a RPIzW and I changed with a RPIZ3B + and I do not have this problem anymore.
What is unfortunate is that bCNC does not keep the coordinates of the axes after restarting the PI and you have to play patience to find a position and resume work. Putting the positions in a backup file and opening this file at startup would be a not insignificant.

Anyway I want to thank you for your ability to share and your kindness to respond to people in trouble.
Plus, of all the software I've tested, it's by far the best and easiest to pick up.

**Comment by Harvie** [comment]
As far as i know, the coordinates are stored in GRBL, not in bCNC

**Comment by JLNapo** [comment]
But is it in bCNC that we set the 3 axes for work? If I am in a position that I stop the application and restart the application, the data are wrong, is it logical?
On a large CNC I use Mach3 on PC Windows and memorization is done (well it's not GRBL either)
But overall, bCNC is doing its job well anyway, and just for that, Bravo!

**Comment by Harvie** [comment]
No. It's always GRBL who remembers the last position. bCNC just shows you state which GRBL reports.
Also if you shut down GRBL, you will always loose few microsteps, because stepper drivers have no memory.

**Comment by JLNapo** [comment]
Ok, thank you very much for these explanations.
Too bad that GRBL is cramped in an ATmega328p, and what do you think cards equipped with ATmega2560 (more space and more ram) but what is it really?

**Comment by Harvie** [comment]
I never really got how this works. For me GRBL sometimes remembers the coordinates and sometimes not. I think problem is that they can't write the memory too often in order to preserve it's life expectancy. Not sure.

**Comment by JLNapo** [comment]
I also think like you.
But could not we get around the problem by storing the information with bCNC and force these parameters on startup? (yes I am heavy)

**Comment by Harvie** [comment]
I don't think it's good idea to force it automaticaly on the startup. But i think we might add some option to store coordinates manualy. Another option might be to develop feature with GRBL developers which will enable use to force storing of coordinates to GRBL memory and call this just before disconnecting/reseting arduino.

**Comment by JLNapo** [comment]
It would be really great!
But the little remaining resource in the atmega328 makes me doubt the possibility.
However on an atmega2560 there is plenty of room, there is a version on github with this processor, and I think the future GRBL is with him.
